python/private/pypi/uv_toolchain.bzl

```diff
@@ -0,0 +1,66 @@
+# Copyright 2024 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module implements the uv toolchain rule"""
+
+UvToolchainInfo = provider(
+    doc = "Information about how to invoke the uv executable.",
+    fields = {
+        "binary": "uv binary",
+        "version": "uv version",
+    },
+)
+
+def _uv_toolchain_impl(ctx):
+    binary = ctx.executable.uv
+
+    templatevariableinfo = platform_common.TemplateVariableInfo({
```

Comment:
I hear you, but I can think some folks who will want to use a `genrule` or something.

My view when shipping toolchains is that they should be as "complete" and flexible as possible, because we don't know how downstream users will need to use them. Even if it's against our best-judgement. I think in this instance, its not any additional maintainer effort to provide this functionality, even if we think users might shoot themselves in the foot.